tests/test_pure.rb

```diff
@@ -85,4 +85,55 @@ def test_reactor_running
     assert a
   end
 
+  module TLSServer
```

Comment:
Would it make sense to use the common ssl tests but with the Pure Ruby reactor under the hood, rather than a separate set of tests?